packages/core/src/modules/credentials/repository/CredentialExchangeRecord.ts

```diff
@@ -80,11 +80,11 @@ export class CredentialExchangeRecord extends BaseRecord<
       this.createdAt = props.createdAt ?? new Date()
       this.state = props.state
       this.connectionId = props.connectionId
-      this.threadId = props.threadId
+      this.threadId = props.threadId ?? ''
```

Comment:
```suggestion
      this.threadId = props.threadId
```

Reply:
note that threadId is now optional in the props, which was a change suggested by @blu3beri.

(it has to be if we create the credential record before the message for the createProposal)